src/realm/util/file.cpp

```diff
@@ -1293,6 +1305,34 @@ bool DirScanner::next(std::string& name)
     }
 }
 
+#elif REALM_HAVE_STD_FILESYSTEM
+
+DirScanner::DirScanner(const std::string& path, bool allow_missing)
+{
+    try {
+        m_iterator = std::filesystem::directory_iterator(path);
+    }
+    catch (const std::filesystem::filesystem_error& e) {
+        static std::error_condition no_such_directory(std::errc::no_such_file_or_directory);
+        if (e.code() != no_such_directory || !allow_missing)
```

Comment:
`e.code()` should be directly comparable with `std::errc::no_such_file_or_directory`, no reason to declare a static local variable. (If it turns out it is necessary after all, that static variable should at least be `const` as well.)